src/ast/kotlin/ast.kt

```diff
@@ -280,31 +280,73 @@ class EnumEntry(override var name: String, val value: String? = null) : Member,
 }
 
 class HeritageType(override var name: String, val needParens: Boolean = false) : Named, Node() {
-    override fun stringify(): String = "$escapedName" + if (needParens) "()" else ""
+    override fun stringify(): String = escapedName + if (needParens) "()" else ""
+}
+
+/**
+ * A reference to a type such as String or Map<String,List<T>> such as for a variable, function param or return type.
+ * @param name the name of the type such as String or Map
+ * @param typeParams the type params such as [String, List<T>], defaulting to empty
+ * @param comment a comment about the type, currently used to document intersection types, defaulting to null.
+ */
+class Type(override var name: String, val typeParams: List<Type> = emptyList(), val comment: String? = null,
```

Comment:
Looks like you can make it data class and remove hand written methods `copy`, `equals` and `hashCode`.